src/lxml/includes/etree_defs.h

```diff
@@ -267,6 +267,12 @@ static void* lxml_unpack_xmldoc_capsule(PyObject* capsule, int* is_owned) {
     xmlDoc *c_doc;
     void *context;
     *is_owned = 0;
+    if (unlikely_condition(!PyCapsule_IsValid(capsule, (const char*)"libxml2:xmlDoc"))) {
```

Comment:
There are two possibilities, the object is not a capsule, i.e. PyCapsule_CheckExact fails or the capsule name is not correct. The documentation of GetPointer only claims that it checks the latter, not the former.